File: Loading Views/[Objective-C] LoadingViews/LoadingViews/MainTableViewController.m
//
//  MainTableViewController.m
//  LoadingViews
//
//  Created by Luis Castillo on 12/24/15.
//  Copyright © 2015 DevBros. All rights reserved.
//

#import "MainTableViewController.h"
#import "MainTableViewCell.h"

#import "ContainerVCViewController.h"
#import "SegmentedMainViewController.h"

@interface MainTableViewController ()

@end

@implementation MainTableViewController


    //MARK: View Loading

- (void)viewDidLoad {
    [super viewDidLoad];
    
    // Uncomment the following line to preserve selection between presentations.
    // self.clearsSelectionOnViewWillAppear = NO;
    
    // Uncomment the following line to display an Edit button in the navigation bar for this view controller.
    // self.navigationItem.rightBarButtonItem = self.editButtonItem;
    
    options = [[NSArray alloc]initWithObjects:@"VC with containers", @"VC in Segmented Controls", nil];
    
}//eom

    //MARK: Memory

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - Table view data source

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {

    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return options.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView
         cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    
    static NSString* cellID = @"myCell";
     MainTableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:cellID forIndexPath:indexPath];
    
    
    int index = (int)indexPath.row;
    
    NSString* currOption = options[index];
    cell.optionName.text = currOption;
    
    return cell;
}//eom

-(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
   int index = (int)indexPath.row;
    
    NSLog(@"option %d selected",index);
    switch (index) {
        case 2:
            break;
        case 1:
            [self GoToSegmentedControl];
            break;
        case 0:
            [self GoToContainersVC];
            break;
                //
        default:
            break;
    }//eo-switch
    
}//eom
/*
// Override to support conditional editing of the table view.
- (BOOL)tableView:(UITableView *)tableView 
 canEditRowAtIndexPath:(NSIndexPath *)indexPath {
    // Return NO if you do not want the specified item to be editable.
    return YES;
}
*/

/*
// Override to support editing the table view.
- (void)tableView:(UITableView *)tableView 
 commitEditingStyle:(UITableViewCellEditingStyle)editingStyle forRowAtIndexPath:(NSIndexPath *)indexPath {
    if (editingStyle == UITableViewCellEditingStyleDelete) {
        // Delete the row from the data source
        [tableView deleteRowsAtIndexPaths:@[indexPath] withRowAnimation:UITableViewRowAnimationFade];
    } else if (editingStyle == UITableViewCellEditingStyleInsert) {
        // Create a new instance of the appropriate class, insert it into the array, and add a new row to the table view
    }   
}
*/

/*
// Override to support rearranging the table view.
- (void)tableView:(UITableView *)tableView 
 moveRowAtIndexPath:(NSIndexPath *)fromIndexPath toIndexPath:(NSIndexPath *)toIndexPath {
}
*/

/*
// Override to support conditional rearranging of the table view.
- (BOOL)tableView:(UITableView *)tableView 
 canMoveRowAtIndexPath:(NSIndexPath *)indexPath {
    // Return NO if you do not want the item to be re-orderable.
    return YES;
}
*/

#pragma mark - Navigation

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
    
}//eom

-(void)GoToContainersVC
{
    ContainerVCViewController * containerVC = [self.storyboard instantiateViewControllerWithIdentifier:@"containerVC"];
    
    [self.navigationController pushViewController:containerVC animated:true];
}//eom

-(void)GoToSegmentedControl
{
    SegmentedMainViewController * segmentedContainerVC = [self.storyboard instantiateViewControllerWithIdentifier:@"segmentedContainer"];
    
    [self.navigationController pushViewController:segmentedContainerVC animated:true];
}//eom






     
     
     
     
     
@end

File: Loading Views/[Swift] LoadingViews/LoadingViews/MainTableViewController.swift
//
//  MainTableViewController.swift
//  LoadingViews
//
//  Created by Luis Andres Castillo Hernandez on 11/18/15.
//  Copyright © 2015 devbros. All rights reserved.
//

import UIKit

class MainTableViewController: UITableViewController {

    
    var options:NSArray?
    
    //MARK: View Loading
    override func viewDidLoad() {
        super.viewDidLoad()

        // Uncomment the following line to preserve selection between presentations
        // self.clearsSelectionOnViewWillAppear = false

        // Uncomment the following line to display an Edit button in the navigation bar for this view controller.
        // self.navigationItem.rightBarButtonItem = self.editButtonItem()
        
        options = [ "Push to NIB TableView" ,
                    "Load view inside a View",
                    "(Explicit) Load view from another SB",
                    "(Fancy) Load view from another SB",
                    "Containers",
                    "Pop Over"]
        
    }//eo-view
    
    
    override func viewDidAppear(animated: Bool)
    {
        
    }//eo-view
    
    //MARK: Memory

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    
    
    // MARK: - Navigation
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }

    

    // MARK: - Table view data source

    override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
        let index = indexPath.row;
        
        print("selected row \(index)")
        if(index == 0)
        {
            let NIBController = TableViewController()
            
            self.navigationController?.navigationBar.barTintColor = UIColor.blueColor()
            
            self.navigationController?.pushViewController(NIBController, animated: true)
        }
        else if(index == 1)
        {   
           let secondViewController = self.storyboard?.instantiateViewControllerWithIdentifier("loadViewFromView") as! AViewController
            
            self.navigationController?.navigationBar.barTintColor = UIColor.lightGrayColor()
            
            self.navigationController?.pushViewController(secondViewController, animated: true)
        }
        else if(index == 2)
        {
            
            let aDiffStoryboard: UIStoryboard = UIStoryboard(name: "StoryboardTwo", bundle: nil)
            
            let thirdViewController = aDiffStoryboard.instantiateViewControllerWithIdentifier("viewInDiffSB") as! ADiffSBViewController
            
            self.navigationController?.navigationBar.barTintColor = UIColor.redColor()
            
            //replaces all views
            self.presentViewController(thirdViewController, animated: true, completion: nil)
            
            //pushes view onto stack
//            self.navigationController?.pushViewController(thirdViewController, animated: true)
        }
        else if (index == 3)
        {
            self.performSegueWithIdentifier("fancyVCPush", sender: self)
        }
        else if(index == 4)
        {
            let viewController = self.storyboard?.instantiateViewControllerWithIdentifier("showContainer") as! MainContainerViewController
            
            self.navigationController?.pushViewController(viewController, animated: true)
        }
        else if(index == 5)
        {
            let viewController = self.storyboard?.instantiateViewControllerWithIdentifier("popOverContainer") as! VCThatCallsPopOverViewController
            
            self.navigationController?.pushViewController(viewController, animated: true)
        }

        
        
        
    }//eom
    
    override func numberOfSectionsInTableView(tableView: UITableView) -> Int {
        return 1
    }

    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        
        return (options?.count)!
    }

    
    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let cell:MainTableViewCell = (tableView.dequeueReusableCellWithIdentifier("mainCell", forIndexPath: indexPath) as? MainTableViewCell)!

        let index = indexPath.row
        cell.cellLabel.text = options![index] as? String
        
        return cell
    }


    /*
    // Override to support conditional editing of the table view.
    override func tableView(tableView: UITableView, canEditRowAtIndexPath indexPath: NSIndexPath) -> Bool {
        // Return false if you do not want the specified item to be editable.
        return true
    }
    */

    /*
    // Override to support editing the table view.
    override func tableView(tableView: UITableView, commitEditingStyle editingStyle: UITableViewCellEditingStyle, forRowAtIndexPath indexPath: NSIndexPath) {
        if editingStyle == .Delete {
            // Delete the row from the data source
            tableView.deleteRowsAtIndexPaths([indexPath], withRowAnimation: .Fade)
        } else if editingStyle == .Insert {
            // Create a new instance of the appropriate class, insert it into the array, and add a new row to the table view
        }    
    }
    */

    /*
    // Override to support rearranging the table view.
    override func tableView(tableView: UITableView, moveRowAtIndexPath fromIndexPath: NSIndexPath, toIndexPath: NSIndexPath) {

    }
    */

    /*
    // Override to support conditional rearranging of the table view.
    override func tableView(tableView: UITableView, canMoveRowAtIndexPath indexPath: NSIndexPath) -> Bool {
        // Return false if you do not want the item to be re-orderable.
        return true
    }
    */

}//eoc
